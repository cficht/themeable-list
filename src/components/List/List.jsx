import React from 'react';
import PropTypes from 'prop-types';

const List = ({ characters }) => {
  const characterNodes = characters.map(character => (
    <li key={character.id}>
      <h3>{character.name}</h3>
      <img src={character.image} />
      <p>{character.status}</p>
      <p>{character.species}</p>
    </li>
  ));

  return (
    <ul>
      {characterNodes}
    </ul>
  );
};

List.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    species: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  })).isRequired
};

export default List;
