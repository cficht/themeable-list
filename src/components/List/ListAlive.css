.List {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  text-align: center;
  padding-inline-start: 0em;
  padding: 20px;
  border: 10px solid rgb(137, 191, 209);
  background-color: lightblue;
  list-style-type: none;

  li {
    border: 10px solid rgb(137, 191, 209);
    border-radius: 15px;
    background-color: lightblue;
    margin: 10px;
    padding: 10px;
  }

  img {
    width: 75%;
  }
}