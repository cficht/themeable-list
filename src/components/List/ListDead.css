.List {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  text-align: center;
  padding-inline-start: 0em;
  padding: 20px;
  border: 10px solid rgb(184, 42, 42);
  background-color: red;
  list-style-type: none;

  li {
    border: 10px solid rgb(184, 42, 42);
    border-radius: 15px;
    background-color: red;
    margin: 10px;
    padding: 10px;
    color: white;
  }

  img {
    width: 75%;
  }

  p {
    font-style: italic;
  }
}
