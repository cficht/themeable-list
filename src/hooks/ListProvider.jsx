import React, { createContext, useState, useEffect, useContext } from 'react';
import PropTypes from 'prop-types';

const ListContext = createContext();

export const ListProvider = ({ children }) => {
  const [characters, setCharacters] = useState([]);

  const chararcterTest = [
    {
      id: 1,
      name: 'Rick Sanchez',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg'
    },
    {
      id: 2,
      name: 'Morty Smith',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg'
    },
    {
      id: 3,
      name: 'Summer Smith',
      status: 'Alive',
      species: 'Human',
      image: 'https://rickandmortyapi.com/api/character/avatar/3.jpeg'
    }
  ];

  useEffect(() => {
    setCharacters(chararcterTest);
  }, []);

  return (
    <ListContext.Provider value={{ characters }}>
      {children}
    </ListContext.Provider>
  );
};

ListProvider.propTypes = {
  children: PropTypes.node
};

export const useCharacters = () => {
  const { characters } = useContext(ListContext);
  return characters;
};
