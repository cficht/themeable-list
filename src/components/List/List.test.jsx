import React from 'react';
import { shallow } from 'enzyme';
import List from './List';
import { ListProvider } from '../../hooks/ListProvider';

describe('List component', () => {
  it('renders List', () => {
    const wrapper = shallow(
      <ListProvider>
        <List />
      </ListProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
