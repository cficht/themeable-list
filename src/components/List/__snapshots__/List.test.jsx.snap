// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders List 1`] = `
<ContextProvider
  value={
    Object {
      "characters": Array [],
      "handlePage": [Function],
      "page": 1,
      "status": "alive",
      "toggle": [Function],
    }
  }
>
  <List />
</ContextProvider>
`;
